File: Classes/Structure/Storage/section.vb
''' <summary>
''' DataType representing any number of Item's as a section().
''' </summary>
Public Class section
	Public Shared arrayArticle() As Article
	Private stringPathSection As String
	Private integerArticleCounter As Integer
''' <summary>
''' Empty Sub.
''' </summary>
	Public Sub New()
		
	End Sub
''' <summary>
''' Get/Adjust the path of the section without Initialize() and Load().
''' </summary>
	Public Property SectionPath As String
		Get
			Return stringPathSection
		End Get
		Set (Value As String)
			stringPathSection = Value
		End Set
	End Property
''' <summary>
''' Resize arrayItem and fill with New Item's.
''' </summary>
	Public Sub SectionInitialize()
		integerArticleCounter = 0
		ReDim arrayArticle(System.IO.Directory.GetFiles(stringPathSection, "*.pd", System.IO.SearchOption.TopDirectoryOnly).GetUpperBound(0))
		For Each file As String In System.IO.Directory.GetFiles(stringPathSection, "*.pd", System.IO.SearchOption.TopDirectoryOnly)
			arrayArticle(integerArticleCounter) = New Article
			arrayArticle(integerArticleCounter).ItemPath = file
			integerArticleCounter += 1
		Next
	End Sub
''' <summary>
''' Loads all Item's with Item Info.
''' </summary>
	Public Sub SectionLoad()
		For Each Article As Article In arrayArticle
			Article.ArticleLoad(stringPathSection)
		Next
	End Sub
''' <summary>
''' Makes new Article with the given template. With ArticleSave().
''' </summary>
''' <param name="ArticleInfo">New String() {ItemPath, ItemFile, Quantity, LastSell, LastBuy}</param>
	Public Sub ArticleAdd(ArticleInfo() As String)
		Dim intCheck As Integer = 0
		Do
			If System.IO.File.Exists(stringPathSection & "\" & intCheck & ".pd") = False Then
				Exit Do
			End If
			intCheck += 1
		Loop
		ReDim Preserve arrayArticle(intCheck)
		arrayArticle(arrayArticle.GetUpperBound(0)) = New Article
		arrayArticle(arrayArticle.GetUpperBound(0)).ArticlePath = stringPathSection & "\"
		arrayArticle(arrayArticle.GetUpperBound(0)).ArticleFile = intCheck & ".pd"
		arrayArticle(arrayArticle.GetUpperBound(0)).ItemPath = ArticleInfo(0)
		arrayArticle(arrayArticle.GetUpperBound(0)).ItemFile = ArticleInfo(1)
		arrayArticle(arrayArticle.GetUpperBound(0)).Quantity = CType(ArticleInfo(2), Integer)
		arrayArticle(arrayArticle.GetUpperBound(0)).LastSell = CType(ArticleInfo(3), Integer)
		arrayArticle(arrayArticle.GetUpperBound(0)).LastBuy = CType(ArticleInfo(4), Integer)
		arrayArticle(arrayArticle.GetUpperBound(0)).ArticleSave(stringPathSection)
	End Sub
End Class

File: Game/Classes/Structure/Storage/section.cs
using System;

/// <summary>
/// DataType representing any number of Item's as a section().
/// </summary>
public class section
{
    public article[] arrayArticle;
    string stringPathSection;
    int integerArticleCounter;

    /// <summary>
    /// Get/Adjust the path of the section without Initialize() and Load().
    /// </summary>
    public string SectionPath {
        get { return stringPathSection; }
        set { stringPathSection = value; }
    }

    /// <summary>
    /// Resize arrayItem and fill with New Item's.
    /// </summary>
    public void SectionInitialize()
    {
        integerArticleCounter = 0;
        arrayArticle = new article[System.IO.Directory.GetFiles(stringPathSection, "*.ini", System.IO.SearchOption.TopDirectoryOnly).GetUpperBound(0) + 1];
        foreach (string file in System.IO.Directory.GetFiles(stringPathSection, "*.ini", System.IO.SearchOption.TopDirectoryOnly)) {
            arrayArticle[integerArticleCounter] = new article();
            arrayArticle[integerArticleCounter].ArticlePath = file;

            integerArticleCounter += 1;
        }
    }
    
    /// <summary>
    /// Loads all Item's with Item Info.
    /// </summary>
    public void SectionLoad()
    {
        foreach (article Article in arrayArticle) {
            Article.articleLoad(stringPathSection + "\\");
        }
    }
    
    /// <summary>
    /// Makes new Article with the given template. With ArticleSave().
    /// </summary>
    /// <param name="ArticleInfo">New String() {ItemPath, ItemFile, Quantity, LastSell, LastBuy}</param>
    public void ArticleAdd(string[] ArticleInfo)
    {
        int intCheck = 0;
        do {
            if (System.IO.File.Exists(stringPathSection + "\\" + intCheck + ".ini")) {
            } else {
                break; // TODO: might not be correct. Was : Exit Do
            }
            intCheck += 1;
        } while (true);
        Array.Resize(ref arrayArticle, intCheck + 1);
        arrayArticle[arrayArticle.GetUpperBound(0)] = new article();
        arrayArticle[arrayArticle.GetUpperBound(0)].ArticlePath = stringPathSection + "\\";
        arrayArticle[arrayArticle.GetUpperBound(0)].ArticleFile = intCheck + ".ini";
        arrayArticle[arrayArticle.GetUpperBound(0)].ItemPath = ArticleInfo[0];
        arrayArticle[arrayArticle.GetUpperBound(0)].ItemFile = ArticleInfo[1];
        arrayArticle[arrayArticle.GetUpperBound(0)].Quantity = Convert.ToInt32(ArticleInfo[2]);
        arrayArticle[arrayArticle.GetUpperBound(0)].LastSell = Convert.ToInt32(ArticleInfo[3]);
        arrayArticle[arrayArticle.GetUpperBound(0)].LastBuy = Convert.ToInt32(ArticleInfo[4]);
        arrayArticle[arrayArticle.GetUpperBound(0)].ArticleSave(stringPathSection);
    }
}